src/main/scala/viper/gobra/backend/Silicon.scala

```diff
@@ -23,18 +22,27 @@ class Silicon(commandLineArguments: Seq[String]) extends ViperVerifier {
       val backend: silicon.Silicon = silicon.Silicon.fromPartialCommandLineArguments(commandLineArguments, reporter)
       
       val startTime = System.currentTimeMillis()
-      backend.start()
-      val result = backend.verify(program)
-      backend.stop()
+      try {
+        backend.start()
+        val result = backend.verify(program)
+        backend.stop()
 
-      result match {
-        case Success =>
-          reporter report OverallSuccessMessage(backend.name, System.currentTimeMillis() - startTime)
-        case f@Failure(_) =>
-          reporter report OverallFailureMessage(backend.name, System.currentTimeMillis() - startTime, f)
-      }
+        result match {
+          case Success =>
+            reporter report OverallSuccessMessage(backend.name, System.currentTimeMillis() - startTime)
+          case f@Failure(_) =>
+            reporter report OverallFailureMessage(backend.name, System.currentTimeMillis() - startTime, f)
+        }
 
-      result
+        result
+      } catch {
+        case _: java.lang.UnsatisfiedLinkError => println("Couldn't find Z3 java API. No libz3java in java.library.path")
+          new Failure(Seq.empty)
+        case e: Throwable =>
+          println("Couldn't verify package due to internal error:")
+          println(e.getMessage())
+          new Failure(Seq.empty)
```

Comment:
Is this case required to make the pattern matching in the `catch` block exhaustive? If it is not required, I would drop it and preserve the old behavior for when an exception is thrown. If it is necessary, I would just re-throw the exception without adding additional text